pkg/network/controllers/vm_test.go

```diff
@@ -250,6 +265,114 @@ var _ = Describe("VM Network Controller", func() {
+			VirtualMachineInstances(vmi.Namespace).
+			Get(context.Background(), vmi.Name, k8smetav1.GetOptions{})
+		Expect(err).NotTo(HaveOccurred())
+
+		Expect(updatedVMI.Spec.Domain.Devices.Interfaces).To(
+			Equal(vm.Spec.Template.Spec.Domain.Devices.Interfaces))
+
+		Expect(updatedVMI.Spec.Networks).To(Equal(updatedVM.Spec.Template.Spec.Networks))
+	},
+		Entry("up to up", v1.InterfaceStateLinkUp, v1.InterfaceStateLinkUp),
+		Entry("up to down", v1.InterfaceStateLinkUp, v1.InterfaceStateLinkDown),
+		Entry("up to absent", v1.InterfaceStateLinkUp, v1.InterfaceStateAbsent),
+		Entry("up to empty", v1.InterfaceStateLinkUp, v1.InterfaceState("")),
+		Entry("down to up", v1.InterfaceStateLinkDown, v1.InterfaceStateLinkUp),
+		Entry("down to down", v1.InterfaceStateLinkDown, v1.InterfaceStateLinkDown),
+		Entry("down to absent", v1.InterfaceStateLinkDown, v1.InterfaceStateAbsent),
+		Entry("down to empty", v1.InterfaceStateLinkDown, v1.InterfaceState("")),
+		Entry("empty to up", v1.InterfaceState(""), v1.InterfaceStateLinkUp),
+		Entry("empty to down", v1.InterfaceState(""), v1.InterfaceStateLinkDown),
+		Entry("empty to absent", v1.InterfaceState(""), v1.InterfaceStateAbsent),
+		Entry("empty to empty", v1.InterfaceState(""), v1.InterfaceState("")),
+	)
+
+	DescribeTable("sync doesn't update link state if hot-unplug is underway ", func(toState v1.InterfaceState) {
+		clientset := fake.NewSimpleClientset()
+		c := controllers.NewVMController(
+			clientset,
+			stubPodGetter{pod: &k8sv1.Pod{}},
+		)
+		const defaultNetName = "default"
+		vmi := libvmi.New(
+			libvmi.WithInterface(v1.Interface{
+				Name:  defaultNetName,
+				State: v1.InterfaceStateAbsent,
+				InterfaceBindingMethod: v1.InterfaceBindingMethod{
+					Bridge: &v1.InterfaceBridge{},
+				},
+			}),
+			libvmi.WithNetwork(v1.DefaultPodNetwork()),
+			libvmistatus.WithStatus(
+				libvmistatus.New(libvmistatus.WithInterfaceStatus(
+					v1.VirtualMachineInstanceNetworkInterface{Name: defaultNetName},
+				)),
+			),
+		)
+
+		vm := libvmi.NewVirtualMachine(vmi.DeepCopy())
+
+		_, err := clientset.KubevirtV1().VirtualMachineInstances(vmi.Namespace).Create(context.Background(), vmi, k8smetav1.CreateOptions{})
+		Expect(err).NotTo(HaveOccurred())
+
+		vm.Spec.Template.Spec.Domain.Devices.Interfaces[0].State = toState
+
+		updatedVM, err := c.Sync(vm, vmi)
+		Expect(err).NotTo(HaveOccurred())
+
+		updatedVMI, err := clientset.KubevirtV1().
+			VirtualMachineInstances(vmi.Namespace).
+			Get(context.Background(), vmi.Name, k8smetav1.GetOptions{})
+		Expect(err).NotTo(HaveOccurred())
+
+		Expect(updatedVMI.Spec.Networks).To(Equal(updatedVM.Spec.Template.Spec.Networks))
+
+		Expect(updatedVMI.Spec.Domain.Devices.Interfaces).To(HaveLen(1))
+		Expect(updatedVMI.Spec.Domain.Devices.Interfaces[0].Name).To(
+			Equal(updatedVM.Spec.Template.Spec.Domain.Devices.Interfaces[0].Name))
+		Expect(updatedVMI.Spec.Domain.Devices.Interfaces[0].State).To(Equal(v1.InterfaceStateAbsent))
```

Comment:
Since nothing should change for the VMI:
Expect(updatedVMI.Spec.Domain.Devices.Interfaces).To(Equal(vmi.Spec.Domain.Devices.Interfaces))

Same for its networks.

Reply:
The previous assertions are still here.